src/topology/homotopy/homotopy_group.lean

```diff
@@ -193,3 +343,63 @@ begin
         { convert H.eq_snd _ _, exacts [y.one_char, iH] },
       end }⟩],
 end
+
+section
+variables {n : ℕ} (i : fin (n+1))
+
+/-- Group structure on `π_(n+1)`. -/
+instance group : group (π_(n+1) X x) :=
+(homotopy_group_equiv_fundamental_group 0).group
+
+/-- Another group structure on `π_(n+2)` that distributes over the default one,
+  so as to enable the Eckmann-Hilton argument. -/
+@[reducible] private def aux_group : group (π_(n+2) X x) :=
+(homotopy_group_equiv_fundamental_group 1).group
+
+lemma from_path_trans_to_path {p q : gen_loop N x} (i : N) {t} :
+  (path_equiv i).symm ((path_equiv i p).trans $ path_equiv i q) t = if (t i : ℝ) ≤ 1/2
+    then p (λ j, if j = i then set.proj_Icc 0 1 zero_le_one (2 * t i) else t j)
+    else q (λ j, if j = i then set.proj_Icc 0 1 zero_le_one (2 * t i - 1) else t j) :=
```

Comment:
I wrote this originally but I now think we should probably make the RHS a `def` (maybe called `mul_at i p q`) with continuity transferred from the LHS. Then we can state `mul_spec` (for all `i : N`) using the new `def` rather than using an existential.

Reply:
I've also written a `def` for `symm_spec` in the same vein, I could use some help golfing the proofs.
To state `mul_spec` for all `i : N` we need the group instance using `i` instead of `0 :  fin (n+1)` but using `i` the instance can't be synthesized, so I left it in terms of `0 : fin (n+1)`.

Reply:
I pushed the generalizations to all `i : N` and changed your definitions to use `gen_loop.copy` which I introduced. If you are happy with it, I'll send it to maintainers soon!

Reply:
Looks fine to me 👍

Reply:
Thanks! However as I took a final look I found some more problems in docstrings and went on to improve them. Currently, there is a final [naming issue](https://github.com/leanprover-community/mathlib/pull/15681/files#r1189178897) that remains. Do you think we should change the names?